### DISCUSSION TITLE
Create a rectangle using the bl, br, tl, tr coordinates instead of top, left, width, height?

### DISCUSSION THREAD

**Discussion by advename** [fabricjs/fabric.js#6834] [General]
So, I'm not sure if this already possible, but I couldn't seem to find a method (or a workaround) that creates a rectangle using the four `aCoords` (`bl`, `br`, `tl`, `tr`) coordinates instead of the `top`, `left`, `width`, `height` values?

My take can be found here: https://jsfiddle.net/7neukojd/32/ , but failed when it comes to rotating the rectangle and then using the `aCoords` to recreate it.

**Comment by asturur** [comment]
It can probably be done, but before i would like to ask you which is the final goal. Can you share it? As described seems a narrow part of a different problem and before doing a coding exercise i would like to know if there are simpler answers to your issue.

**Reply by advename** [reply]
The problem i have is that a user has multiple drawing tasks.
Tasks are separated by one task per page.
The user can hit the "next" or "previous" button to navigate between pages.
On a task page,
1.the user draws a rectangle
2. the `bl`, `br`, `tl`, `tr` coordinates are stored in the backend
3. whenever the user navigates to this same page, the page re-creates the rectangle using the `bl`, `br`, `tl`, `tr` coordinates from the backend

**Comment by asturur** [comment]
What is the reason to not store  the `rect.toObject()` specifying `rect.includeDefaultValues = false` That should give you a small enough object and out of the box compatibility to restore it.

I can try to resolve your question anyway.

**Reply by advename** [reply]
This seems to solve the problem too, but would require to re-write multiple backend methods which expected the aCoords, or create a new DB table that stores these values. I.e. that would have been my last resort!

**Comment by asturur** [comment]
https://jsfiddle.net/kwbdrx48/

I think is all about geometry, nothing fabric specific ( apart knowing that a fabric.Point class exists ).

**Reply by advename** [reply]
You are amazing, thank you a lot!

**Edit (removed the code i posted for archive reasons since it was the wrong one)**
Look comment below to see the code if jsFiddle link has been removed!

@asturur, anyway I can buy you a beer/coffee? You have Paypal or something? 
This is the third time you helped me out!

**Reply by asturur** [reply]
> @asturur, anyway I can buy you a beer/coffee? You have Paypal or something?
> This is the third time you helped me out!

If you want you can become a sponsor, there are tiers as low as 1$ a month, but i m not sure what is the minimum amount of months you can subscribe to the sponsorship program.

**Reply by advename** [reply]
Well, I've seen these support links, but i wanted to donate directly to you.
But will do that then.

**Comment by asturur** [comment]
That's not the right code :D

**Reply by advename** [reply]
What do you mean that is not the right code? I copied it straight from your updated jsFiddle? :grimacing:

**Reply by asturur** [reply]
i wrote this:

```js
  let rectObj = canvas1.getObjects()[0];
  const tl = rectObj.aCoords.tl;
  const tr = rectObj.aCoords.tr;
  const bl = rectObj.aCoords.bl;
  
  let rectCopy = new fabric.Rect({
    left: rectObj.aCoords.tl.x,
    top:  rectObj.aCoords.tl.y,
    width: (new fabric.Point(tl.x, tl.y).distanceFrom(tr)),
    height: (new fabric.Point(tl.x, tl.y).distanceFrom(bl)),
    angle: fabric.util.radiansToDegrees(Math.atan2(tr.y - tl.y, tr.x - tl.x)),
    fill: "red",
  })
```

**Reply by advename** [reply]
Damn you are right, my bad! Thanks for the headsup!

**Comment by asturur** [comment]
The github sponsorship goes directly in my bank account, and as today, is the best way to go for both.
You should not pay any fee, i do not pay any fee and no withdrawal / bank transfer fee.
Until microsoft is paying for this service, is definitely better than any other patreon or paypal.

Then i tend to underline, if you really feel like, but also don't bother.

**Reply by advename** [reply]
Ill try to get onto the sponsorship then, thanks!

**Comment by asturur** [comment]
am improved version with better position 
https://stackoverflow.com/questions/66145267/how-to-set-the-4-corners-x-y-coordinates-value-of-a-fabricjs-image/66185338#66185338

**Comment by agilexmarketing** [comment]
Dear Asturur,

How to find the correct coordinates for the bounding box in case of a very specific font as https://www.dafont.com/scriptina.font

In this case the bounding box is much smaller and even top and left coordinates are not correct ones.

Any help would be appreciated :)

**Reply by gloriousjob** [reply]
This seems like a completely different problem than the topic and should be its own topic. That said, you’ll want to provide an example of what you’re seeing incorrect and maybe a mock-up of what correct would look like if you make a new topic.

**Reply by ShaMan123** [reply]
Indeed.
If you want to solve it you must also provide a repro without fabric, using only canvas API and proving it is able to measure the text properly by drawing the bbox. Then you will prove it is a bug in fabric and not an issue with the font so we can proceed to fixing it.
Of course you can prove this by suggesting a fix directly
